example/src/main/java/org/eclipse/hono/vertx/example/base/AbstractHonoSender.java

```diff
@@ -0,0 +1,204 @@
+
+    private RegistrationClient registrationClient;
+    private MessageSender messageSender;
+
+    private AtomicInteger dispositionHandlerCalled = new AtomicInteger(0);
+
+    private boolean eventMode = false;
+
+    public AbstractHonoSender() {
+        honoMessagingClient = new HonoClientImpl(vertx,
+                ConnectionFactoryImpl.ConnectionFactoryBuilder.newBuilder()
+                        .vertx(vertx)
+                        .host(HONO_MESSAGING_HOST)
+                        .port(HONO_MESSAGING_PORT)
+                        .user(HONO_CLIENT_USER)
+                        .password(HONO_CLIENT_PASSWORD)
+                        .trustStorePath("target/config/hono-demo-certs-jar/trusted-certs.pem")
+                        .disableHostnameVerification()
+                        .build());
+        honoRegistryClient = new HonoClientImpl(vertx,
+                ConnectionFactoryImpl.ConnectionFactoryBuilder.newBuilder()
+                        .vertx(vertx)
+                        .host(HONO_REGISTRY_HOST)
+                        .port(HONO_REGISTRY_PORT)
+                        .user(HONO_CLIENT_USER)
+                        .password(HONO_CLIENT_PASSWORD)
+                        .trustStorePath("target/config/hono-demo-certs-jar/trusted-certs.pem")
+                        .disableHostnameVerification()
+                        .build());
+    }
+
+    protected void sendData() throws Exception {
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        final AtomicBoolean clientsCreated = getHonoClients(latch);
+
+        latch.await();
+
+        if (clientsCreated.get()) {
+            IntStream.range(0, COUNT).forEach(value -> {
+                handleSingleMessage(value);
+            });
+        }
+
+        vertx.close();
+    }
+
+    private boolean sendMessageToHono(final int value, final String token) {
+        final Map<String, Object> properties = new HashMap<>();
+        properties.put("my_prop_string", "I'm a string");
+        properties.put("my_prop_int", value);
+
+        if (eventMode) {
+            final BiConsumer<Object, ProtonDelivery> myDispositionHandler = (messageId, delivery) -> {
+                dispositionHandlerCalled.incrementAndGet();
+                if (! (delivery.getRemoteState() instanceof Accepted)) {
+                    System.err.println("delivery failed for [message ID: " + messageId + ", new remote state: " + delivery.getRemoteState());
+                } else {
+                    System.out.println("delivery accepted for [message ID: " + messageId);
+                }
+            };
+            return (messageSender.send(DEVICE_ID, properties, "myMessage" + value, "text/plain", token, myDispositionHandler));
+        } else {
+            return (messageSender.send(DEVICE_ID, properties, "myMessage" + value, "text/plain", token));
+        }
+    }
+
+    private AtomicBoolean getHonoClients(final CountDownLatch countDownLatch) {
```

Comment:
IMHO returning a `Future<Void` would be much easier to implement and deal with. It would also much more easily fit with the `Future.compose()` style of concatenating activities ..

Reply:
+1

Reply:
I was referring to the `getHonoClients` method, which FMPOV should return a `Future<Void>` ...